scripts/utilities.sh

```diff
@@ -0,0 +1,610 @@
+          fi
+          if [ "${group_dialout}" == "false" ]; then
+            sudo usermod -a -G dialout "${USER}" && ok_msg "Group 'dialout' assigned!"
+          fi
+          ok_msg "Remember to relog/restart this machine for the group(s) to be applied!"
+          break;;
+        N|n|No|no)
+          select_msg "No"
+          break;;
+        *)
+          print_error "Invalid command!";;
+      esac
+    done
+  fi
+}
+
+function set_custom_hostname(){
+  echo
+  top_border
+  echo -e "|  Changing the hostname of this machine allows you to  |"
+  echo -e "|  access a webinterface that is configured for port 80 |"
+  echo -e "|  by simply typing '<hostname>.local' in the browser.  |"
+  echo -e "|                                                       |"
+  echo -e "|  E.g.: If you set the hostname to 'my-printer' you    |"
+  echo -e "|        can open Mainsail / Fluidd / Octoprint by      |"
+  echo -e "|        browsing to: http://my-printer.local           |"
+  bottom_border
+  while true; do
+    read -p "${cyan}###### Do you want to change the hostname? (y/N):${white} " yn
+    case "${yn}" in
+      Y|y|Yes|yes)
+        select_msg "Yes"
+        change_hostname
+        break;;
+      N|n|No|no|"")
+        select_msg "No"
+        break;;
+      *)
+        error_msg "Invalid command!";;
+    esac
+  done
+}
+
+function change_hostname(){
+    local new_hostname
+    echo
+    top_border
+    echo -e "|  ${green}Allowed characters: a-z, 0-9 and single '-'${white}          |"
+    echo -e "|  ${red}No special characters allowed!${white}                       |"
+    echo -e "|  ${red}No leading or trailing '-' allowed!${white}                  |"
+    bottom_border
+    while true; do
+      read -p "${cyan}###### Please set the new hostname:${white} " new_hostname
+      if [[ ${new_hostname} =~ ^[^\-\_]+([0-9a-z]\-{0,1})+[^\-\_]+$ ]]; then
+        while true; do
+          echo
+          read -p "${cyan}###### Do you want '${new_hostname}' to be the new hostname? (Y/n):${white} " yn
+          case "${yn}" in
+            Y|y|Yes|yes|"")
+              select_msg "Yes"
+              set_hostname "${new_hostname}"
+              break;;
+            N|n|No|no)
+              select_msg "No"
+              abort_msg "Skip hostname change ..."
+              break;;
+            *)
+              print_error "Invalid command!";;
+          esac
+        done
+      else
+        warn_msg "'${new_hostname}' is not a valid hostname!"
+      fi
+      break
+    done
+}
+
+function set_hostname(){
+  local new_hostname=${1} current_date
+  #check for dependencies
+  local dep=(avahi-daemon)
+  dependency_check "${dep[@]}"
+
+  #create host file if missing or create backup of existing one with current date&time
+  if [ -f /etc/hosts ]; then
+    current_date=$(get_date)
+    status_msg "Creating backup of hosts file ..."
+    sudo cp "/etc/hosts /etc/hosts.${current_date}.bak"
```

Comment:
```suggestion
    sudo cp "/etc/hosts" "/etc/hosts.${current_date}.bak"
```

Reply:
I had to make this change in order to fix the error I got when I attempted to set the hostname on armbian-22.05-trunk.

```
###### Creating backup of hosts file ...
cp: missing destination file operand after '/etc/hosts /etc/hosts.220503-2314.bak'
Try 'cp --help' for more information.
```

Reply:
Thank you! Yes that was indeed a typo and i just fixed it.